test/celeritas/decay/MuDecay.test.cc

```diff
@@ -0,0 +1,139 @@
+
+TEST_F(MuDecayInteractorTest, basic)
+{
+    auto const& params = *this->particle_params();
+    auto const at_rest = MevEnergy{0};
+    auto const max_lepton_energy = real_type{0.5} * data_.muon_mass.value()
+                                   - data_.electron_mass.value();
+
+    // Anti-muon decay
+    {
+        this->set_inc_particle(pdg::mu_plus(), at_rest);
+        MuDecayInteractor interact(data_,
+                                   this->particle_track(),
+                                   this->direction(),
+                                   this->secondary_allocator());
+        auto result = interact(this->rng());
+
+        EXPECT_EQ(Interaction::Action::decay, result.action);
+        auto const& sec = result.secondaries;
+        EXPECT_EQ(1, sec.size());
+        EXPECT_EQ(pdg::positron(), params.id_to_pdg(sec[0].particle_id));
+        EXPECT_GE(max_lepton_energy, sec[0].energy.value());
+    }
+
+    // Muon decay
+    {
+        this->set_inc_particle(pdg::mu_minus(), at_rest);
+        MuDecayInteractor interact(data_,
+                                   this->particle_track(),
+                                   this->direction(),
+                                   this->secondary_allocator());
+        auto result = interact(this->rng());
+
+        EXPECT_EQ(Interaction::Action::decay, result.action);
+        auto const& sec = result.secondaries;
+        EXPECT_EQ(1, sec.size());
+        EXPECT_EQ(pdg::electron(), params.id_to_pdg(sec[0].particle_id));
+        EXPECT_GE(max_lepton_energy, sec[0].energy.value());
+    }
+}
+
+//---------------------------------------------------------------------------//
+TEST_F(MuDecayInteractorTest, stress_test)
+{
+    size_type const num_secondaries = 1;
+    size_type const num_samples = 10000;
+    MevEnergy const one_gev{1000};
+
+    this->resize_secondaries(num_secondaries * num_samples);
+    this->set_inc_particle(pdg::mu_minus(), one_gev);
+    MuDecayInteractor interact(data_,
+                               this->particle_track(),
+                               this->direction(),
+                               this->secondary_allocator());
+
+    double avg_sec_energies[num_secondaries]{};  // Avg energy per secondary
+    Real3 avg_total_momentum{};  // Avg total momentum per decay
+
+    for ([[maybe_unused]] auto sample : range(num_samples))
+    {
+        auto result = interact(this->rng());
+        auto const& sec = result.secondaries;
+
+        for (auto i : range(sec.size()))
+        {
+            auto const& part = sec[i];
+            avg_sec_energies[i] += part.energy.value();
+            for (auto j : range(3))
+            {
+                avg_total_momentum[j] += part.direction[j]
+                                         * part.energy.value();
+            }
+        }
+    }
+
+    for (auto j : range(num_secondaries))
+    {
+        avg_sec_energies[j] /= num_samples;
```

Comment:
If you include `ArrayOperators` you can just do `avg_sec_energies /= num_samples`